### ISSUE TITLE
investigate high memory consumption

### ISSUE THREAD

**Issue by jyeshe** [aeternity/ae_mdw#414]
There was a growing memory consumption during 1 day while syncing:
![image](https://user-images.githubusercontent.com/44991200/147081731-0960a282-ec7d-4aed-8451-cb44138fc278.png)

**Comment by jyeshe** [comment]
Investigation needed for versions before and after 1.4.0.
Before (multiple times) because:
- ca7f295 was not merged yet (was on hold)
- during the sync process there is no buffer except for caches that were not changed in 1.4.0
- caches were not changed for a long time except for removing them with new pagination
- the latest cached change was`:ets.insert(:ct_create_sync_cache, {contract_pk, txi})` but it is erased every generation

After 1.4.0 (focused on consistency) once:
- this change gonna help releasing memory (mutations outside of the reduce accumulator):
https://github.com/aeternity/ae_mdw/blob/9b1ba2ebf1d03fc2245abfa1750d363ccf013bd7/lib/ae_mdw/db/sync/transaction.ex#L211
- a whole profiling is needed including endpoints (pagination)

**Comment by jyeshe** [comment]
Investigation needed for versions before and after 1.4.0.
Before (multiple times) because:
- during the sync process there is no buffer except for caches that were not changed in 1.4.0
- caches were not changed for a long time except for removing them with new pagination
- the latest cached change was`:ets.insert(:ct_create_sync_cache, {contract_pk, txi})` but it is erased every generation

After 1.4.0 (focused on consistency) once:
- this change gonna help releasing memory (mutations outside of the reduce accumulator):
https://github.com/aeternity/ae_mdw/blob/9b1ba2ebf1d03fc2245abfa1750d363ccf013bd7/lib/ae_mdw/db/sync/transaction.ex#L211
- a whole profiling is needed including endpoints (pagination)

**Comment by thepiwo** [comment]
from my side this was not seen with sync of 1.4.0

**Comment by jyeshe** [comment]
This will be resumed after #529

**Comment by thepiwo** [comment]
I didn't see this behaviour any more

**Comment by jyeshe** [comment]
I wanted to do some profiling after full RocksDB migration but better close it when reopen if any leak is found. Thank you!
Closed by #390
